src/Sylius/Bundle/CoreBundle/Form/Type/Pricing/CurrencyBasedConfigurationType.php

```diff
@@ -0,0 +1,68 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sylius\Bundle\CoreBundle\Form\Type\Pricing;
+
+use Sylius\Component\Resource\Repository\RepositoryInterface;
+use Symfony\Component\Form\AbstractType;
+use Symfony\Component\Form\FormBuilderInterface;
+use Symfony\Component\OptionsResolver\OptionsResolverInterface;
+
+/**
+ * @author Alexandre Bacco <alexandre.bacco@gmail.com>
+ */
+class CurrencyBasedConfigurationType extends AbstractType
+{
+    protected $currencyRepository;
+
+    /**
+     * @param RepositoryInterface $groupRepository
+     */
+    public function __construct(RepositoryInterface $currencyRepository)
+    {
+        $this->currencyRepository = $currencyRepository;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function buildForm(FormBuilderInterface $builder, array $options)
+    {
+        foreach ($this->currencyRepository->findAll() as $currency) {
```

Comment:
Not sure, the provider only provides enabled currencies. And you may want to fill in all prices for a currency *before* enabling it.